tools/serverpod_cli/lib/src/config/config.dart

```
@@ -259,13 +259,10 @@ class GeneratorConfig implements ModelLoadConfig {
     var name = _stripPackage(serverPackage);
 
     var file = File(p.join(serverRootDir, 'config', 'generator.yaml'));
-    Map generatorConfig = {};
-    try {
-      var yamlStr = file.readAsStringSync();
-      generatorConfig = loadYaml(yamlStr);
-    } catch (_) {}
-
-    PackageType type = getPackageType(generatorConfig);
+    YamlMap generatorConfig = await file.exists()
+        ? loadYaml(await file.readAsString(), sourceUrl: file.uri) as YamlMap
```

Question: Aren't there other reasons that this might crash than simply handling if the file doesn't exist? For example if the `loadYaml` fails or if the `cast` is invalid?

---

I want us to fail if we cannot parse the pubspec. Previously we just silently ignored that we had no clue, if the versions where correct.

---

That makes sense! What would the formatted output be to the user? I think we should inform them in a meaningful way that the file is corrupted and that we can't read it.

---

Assuming a `config/generator.yaml` containing just:
```
invalid yaml
```

In 2.4 we hapily ignore any parse errors:
```
serverpod generate 
✓ Generating code (10.0s)
✅ Done.
```
going forward we will error out like:
```
serverpod generate
Building package executable... (12.6s)
Built serverpod_cli:serverpod_cli.
An error occurred while parsing the server config file: Error on line 1, column 1 of config/generator.yaml: Expected a map
  ╷
1 │ invalid yaml
  │ ^^^^^^^^^^^^
  ╵
```